src/key-path.js

```diff
@@ -6,9 +6,36 @@ var isFunction = require('./utils').isFunction
  * @param {*} toTest
  * @return {boolean}
  */
-exports.isKeyPath = function(toTest) {
+var isKeyPath = exports.isKeyPath = function isKeyPath(toTest) {
   return (
     isArray(toTest) &&
     !isFunction(toTest[toTest.length - 1])
   )
 }
+
+/**
+ * Determines if two keyPaths are the same
+ * @param {array} keyPath
+ * @param {array} otherKeyPath
+ * @return {boolean}
+ */
+exports.eql = function eq(keyPath, otherKeyPath) {
+  if (!isKeyPath(keyPath) || !isKeyPath(otherKeyPath)) {
+    return false
+  }
+
+  var len
+  var i = len = keyPath.length
+
+  if (len !== otherKeyPath.length) {
+    return false
+  }
+
+  while (i--) {
+    if (keyPath[i] !== otherKeyPath[i]) {
```

Comment:
I might be missing something, but this is just for comparing array keypaths of `Array [ String, ...]`. It uses `isKeyPath` to make sure of that.

The actual getter comparison is an `===` reference check, since we have access to the observer entry: https://github.com/optimizely/nuclear-js/pull/115/files#diff-889e806243cf10b2e10689e9632f17a2R132.

[Here](https://github.com/optimizely/nuclear-js/pull/115/files#diff-7e02ba8688bee6af79401977651e94b9R284)'s the test for getters, using `totalGetter`, which is composed of other getters.